### PR TITLE
Add password as a separate property in unregistered user

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

In the new registration flow, the password will not be provided at the same time as the email. Therefore, we cannot use the previous "credentials" structure as is, because it requires to store the password at the same time as the email.

### Solutions

Add the password as a separate property, and add an helper property to check if the user needs a password.

### Notes

Do not merge this PR yet, as it requires changes in SE and UI.

### CODE REVIEW

File: Source/Model/User/UnregisteredUser.swift [hunk 1]
```diff
@@ -71,6 +80,7 @@ extension UnregisteredUser: Equatable {
             && lhs.accentColorValue == rhs.accentColorValue
             && lhs.acceptedTermsOfService == rhs.acceptedTermsOfService
             && lhs.marketingConsent == rhs.marketingConsent
+            && lhs.password == rhs.password
```
Comment (vytis) [inline Source/Model/User/UnregisteredUser.swift]: Can we use automatically-derived `Equatable` here?

Comment (alexaubry) [reply Source/Model/User/UnregisteredUser.swift]: I think not, because it's a class?

Comment (vytis) [reply Source/Model/User/UnregisteredUser.swift]: 😿

**Comment by alexaubry** [comment]
@zenkins release major

**Pull request merged**